tests/jvm/src/test/scala/munit/AsyncFixtureOrderSuite.scala

```diff
@@ -0,0 +1,38 @@
+package munit
+
+import scala.concurrent.Future
+import scala.concurrent.Promise
+
+class AsyncFixtureOrderSuite extends FunSuite {
+  val latch = Promise[Unit]
+  var completedFromTest = Option.empty[Boolean]
+  var completedFromTeardown = Option.empty[Boolean]
+
+  val latchOnTeardown = FunFixture.async[String](
+    setup = { test => Future.successful(test.name) },
+    teardown = { name =>
+      implicit val ec = munitExecutionContext
+      Future {
+        completedFromTeardown = Some(latch.trySuccess(()));
+      }
+    }
+  )
+
+  override def afterAll(): Unit = {
+    // promise was completed first form the test
+    assertEquals(completedFromTest, Some(true))
+    // and then from the teardown
```

Comment:
To my understanding, the thing about Promises is that they can be completed
only _once_.  Do you mean something like
"and not by the teardown"?

Reply:
Yup, you are right. I will reword. This test asserts completion *attempts* of that promise. Asserting on those booleans allows to track the order of promise completion attempts.